ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -278,11 +278,16 @@ OptimizedConvBlockConfig determine_per_core_conv_block_config(
     uint32_t window_w,
     bool fp32_accum,
     bool use_shallow_conv_variant) {
+
     if (act_block_h_override > 0) {
         TT_ASSERT(
             act_block_h_override % 32 == 0,
             "Config Error: act_block_h_override must be a multiple of 32 (tile height).");
     }
+    if(parallel_config.shard_scheme == TensorMemoryLayout::WIDTH_SHARDED)
```

Comment:
uint32_t act_block_h_ntiles = conv_op_parallel_config.per_core_out_matrix_height_ntiles;
if (parallel_config.shard_scheme != TensorMemoryLayout::WIDTH_SHARDED && act_block_h_override > 0) {
    act_block_h_ntiles = act_block_h_override / 32;
}

Can we use something like this?
And put a comment on act_block_h_override in Conv2dConfig that act_block_h_override is not applied in case of Width sharding?

Reply:
Should we log at least at the info level (not sure if warning is too much) that this is happening and the value should be changed?

Also, whatever code path is taken for the "auto sharding" feature, it should choose a proper config value if it doesn't already.

(moved discussion over from https://github.com/tenstorrent/tt-metal/pull/13941/files#r1804983425)

Reply:
Updated to use the same approach.

Reply:
Added logging.